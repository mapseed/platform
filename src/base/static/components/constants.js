const constants = {
  TEXT_FIELD_TYPENAME: "text",
  TEXTAREA_FIELD_TYPENAME: "textarea",
  RICH_TEXTAREA_FIELD_TYPENAME: "rich_textarea",
  BIG_CHECKBOX_FIELD_TYPENAME: "big_checkbox",
  BIG_RADIO_FIELD_TYPENAME: "big_radio",
  DROPDOWN_FIELD_TYPENAME: "dropdown",
  DROPDOWN_AUTOCOMPLETE_FIELD_TYPENAME: "dropdown_autocomplete",
  PUBLISH_CONTROL_TOOLBAR_TYPENAME: "publish_control_toolbar",
  MAP_DRAWING_TOOLBAR_TYPENAME: "map_drawing_toolbar",
  CUSTOM_URL_TOOLBAR_TYPENAME: "custom_url_toolbar",
  DATETIME_FIELD_TYPENAME: "datetime",
  GEOCODING_FIELD_TYPENAME: "geocoding",
  BIG_TOGGLE_FIELD_TYPENAME: "big_toggle",
  ATTACHMENT_FIELD_TYPENAME: "file",
  COMMON_FORM_ELEMENT_TYPENAME: "common_form_element",
  SUBMIT_FIELD_TYPENAME: "submit",
  RANGE_FIELD_TYPENAME: "range",

  AUTOFILL_DURATION_DAYS: 30,

  RICH_TEXT_IMAGE_MARKUP_PREFIX: "{{#rich-text-image ",
  RICH_TEXT_IMAGE_MARKUP_SUFFIX: "}}",

  FIELD_STATE_VALUE_KEY: "value",
  FIELD_STATE_VALIDITY_KEY: "isValid",
  FIELD_STATE_VALIDITY_MESSAGE_KEY: "message",
  FIELD_STATE_FIELD_TYPE_KEY: "type",
  FIELD_STATE_INITIALIZED_KEY: "isInitialized",

  ON_ADD_ATTACHMENT_ACTION: "on-add-attachment",
};

export default constants;
