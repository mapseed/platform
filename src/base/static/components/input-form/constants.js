const constants = {
  TEXT_FIELD_TYPENAME: "text",
  TEXTAREA_FIELD_TYPENAME: "textarea",
  RICH_TEXTAREA_FIELD_TYPENAME: "rich_textarea",
  BIG_CHECKBOX_FIELD_TYPENAME: "big_checkbox",
  BIG_RADIO_FIELD_TYPENAME: "big_radio",
  DROPDOWN_FIELD_TYPENAME: "dropdown",
  DROPDOWN_AUTOCOMPLETE_FIELD_TYPENAME: "dropdown_autocomplete",
  PUBLISH_CONTROL_TOOLBAR_TYPENAME: "publish_control_toolbar",
  MAP_DRAWING_TOOLBAR_TYPENAME: "map_drawing_toolbar",
  CUSTOM_URL_TOOLBAR_TYPENAME: "custom_url_toolbar",
  DATETIME_FIELD_TYPENAME: "datetime",
  GEOCODING_FIELD_TYPENAME: "geocoding",
  BIG_TOGGLE_FIELD_TYPENAME: "big_toggle",
  ATTACHMENT_FIELD_TYPENAME: "file",
  COMMON_FORM_ELEMENT_TYPENAME: "common_form_element",
  SUBMIT_FIELD_TYPENAME: "submit"
}

export default constants;
