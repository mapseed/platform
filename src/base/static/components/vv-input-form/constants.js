export default {
  COMMUNITY_INPUT_CATEGORY_NAME: "community_input",
  INPUT_CATEGORY_FIELDNAME: "input_category",
  INPUT_SUBCATEGORY_FIELDNAME: "input_subcategory",
  INPUT_CATEGORY_SUMMARY_NAME: "summary",
  INPUT_SUBCATEGORY_SUMMARY_NAME: "all",
  IS_STICKY_FIELDNAME: "is_sticky",
  CREATED_DATETIME_FIELDNAME: "created_datetime",
  INPUT_TEXT_FIELDNAME: "input_text",
  RECOMMENDATION_CATEGORY_NAME: "recommendation",
  CONCERN_CATEGORY_NAME: "concern",
};
