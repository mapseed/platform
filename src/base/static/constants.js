export default {
  TEXT_FIELD_TYPENAME: "text",
  TEXTAREA_FIELD_TYPENAME: "textarea",
  RICH_TEXTAREA_FIELD_TYPENAME: "rich_textarea",
  BIG_CHECKBOX_FIELD_TYPENAME: "big_checkbox",
  BIG_RADIO_FIELD_TYPENAME: "big_radio",
  DROPDOWN_FIELD_TYPENAME: "dropdown",
  DROPDOWN_AUTOCOMPLETE_FIELD_TYPENAME: "dropdown_autocomplete",
  PUBLISH_CONTROL_TOOLBAR_TYPENAME: "publish_control_toolbar",
  MAP_DRAWING_TOOLBAR_TYPENAME: "map_drawing_toolbar",
  CUSTOM_URL_TOOLBAR_TYPENAME: "custom_url_toolbar",
  DATETIME_FIELD_TYPENAME: "datetime",
  GEOCODING_FIELD_TYPENAME: "geocoding",
  BIG_TOGGLE_FIELD_TYPENAME: "big_toggle",
  ATTACHMENT_FIELD_TYPENAME: "file",
  COMMON_FORM_ELEMENT_TYPENAME: "common_form_element",
  SUBMIT_FIELD_TYPENAME: "submit",
  RANGE_FIELD_TYPENAME: "range",
  INFORMATIONAL_HTML_FIELD_TYPENAME: "informational_html",

  AUTOFILL_DURATION_DAYS: 30,

  LOCATION_TYPE_PROPERTY_NAME: "location_type",
  TITLE_PROPERTY_NAME: "title",
  FULL_TITLE_PROPERTY_NAME: "fullTitle",
  NAME_PROPERTY_NAME: "name",
  SUBMITTER_NAME: "submitter_name",
  SUBMITTER: "submitter",
  USER_TOKEN_PROPERTY_NAME: "user_token",
  CREATED_DATETIME_PROPERTY_NAME: "created_datetime",
  DATASET_SLUG_PROPERTY_NAME: "datasetSlug",
  MODEL_ID_PROPERTY_NAME: "id",
  STORY_FIELD_NAME: "story",
  DATASET_ID_PROPERTY_NAME: "datasetId",
  CUSTOM_URL_PROPERTY_NAME: "url-title",
  SHOW_METADATA_PROPERTY_NAME: "showMetadata",
  IS_VISIBLE_PROPERTY_NAME: "visible",
  IS_FOCUSED_PROPERTY_NAME: "isFocused",

  ATTACHMENT_TYPE_PROPERTY_NAME: "type",
  ATTACHMENT_NAME_PROPERTY_NAME: "name",
  ATTACHMENT_FILE_PROPERTY_NAME: "file",
  RICH_TEXT_IMAGE_CODE: "RT",
  COVER_IMAGE_CODE: "CO",

  RICH_TEXT_IMAGE_MARKUP_PREFIX: "{{#rich-text-image ",
  RICH_TEXT_IMAGE_MARKUP_SUFFIX: "}}",

  FIELD_VALUE_KEY: "value",
  FIELD_VALIDITY_KEY: "isValid",
  FIELD_VALIDITY_MESSAGE_KEY: "message",
  FIELD_FIELD_TYPE_KEY: "type",
  FIELD_RENDER_KEY: "renderKey",
  FIELD_VISIBILITY_KEY: "isVisible",
  FIELD_TRIGGER_VALUE_KEY: "trigger",
  FIELD_TRIGGER_TARGETS_KEY: "triggerTargets",
  FIELD_AUTO_FOCUS_KEY: "isAutoFocusing",
  FIELD_ADVANCE_STAGE_ON_VALUE_KEY: "advanceStage",

  GEOMETRY_PROPERTY_NAME: "geometry",
  GEOMETRY_STYLE_PROPERTY_NAME: "style",
  GEOMETRY_TYPE_PROPERTY_NAME: "type",

  // These geometry style property names follow the SimpleStyle spec.
  // https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0
  LINE_COLOR_PROPERTY_NAME: "stroke",
  LINE_OPACITY_PROPERTY_NAME: "stroke-opacity",
  FILL_COLOR_PROPERTY_NAME: "fill",
  FILL_OPACITY_PROPERTY_NAME: "fill-opacity",
  MARKER_ICON_PROPERTY_NAME: "marker-symbol",

  DRAW_DEFAULT_LINE_COLOR: "#f86767",
  DRAW_DEFAULT_LINE_OPACITY: 0.7,
  DRAW_DEFAULT_FILL_COLOR: "#f1f075",
  DRAW_DEFAULT_FILL_OPACITY: 0.3,
  DRAW_CREATE_MARKER_TOOL: "create-marker",
  DRAW_CREATE_POLYLINE_TOOL: "create-polyline",
  DRAW_CREATE_POLYGON_TOOL: "create-polygon",
  DRAW_DELETE_GEOMETRY_EVENT: "draw:delete",
  DRAW_START_POLYGON_EVENT: "draw:start-polygon",
  DRAW_START_POLYLINE_EVENT: "draw:start-polyline",
  DRAW_START_MARKER_EVENT: "draw:start-marker",
  DRAW_UPDATE_GEOMETRY_EVENT: "draw:update-geometry",
  DRAW_INIT_GEOMETRY_EVENT: "draw:init-geometry",
  DRAW_STROKE_COLORPICKER_NAME: "stroke",
  DRAW_FILL_COLORPICKER_NAME: "fill",
  DRAW_MARKER_SELECTOR_NAME: "marker",

  PLACE_MODEL_UPDATE_EVENT: "place-model:update",
  PLACE_MODEL_REMOVE_EVENT: "place-model:remove",

  IS_HIDDEN_BY_FILTERS: "_mapseed:isHiddenByFilters",
  IS_FOCUSED_BY_FILTERS: "_mapseed:isFocusedByFilters",

  PLACE_COLLECTION_UNFOCUS_ALL_PLACES_EVENT:
    "place-collection:unfocus-all-places",
  PLACE_COLLECTION_HIDE_PLACE_EVENT: "place-collection:hide-place",
  PLACE_COLLECTION_FOCUS_PLACE_EVENT: "place-collection:focus-place",
  PLACE_COLLECTION_ADD_PLACE_EVENT: "place-collection:add-place",
  PLACE_COLLECTION_LOADED_EVENT: "place-collection:loaded",

  PLACE_MODEL_IO_START_ACTION: "place:io-start",
  PLACE_MODEL_IO_END_SUCCESS_ACTION: "place:io-end-success",
  PLACE_MODEL_IO_END_ERROR_ACTION: "place:io-end-error",
  SURVEY_MODEL_IO_END_SUCCESS_ACTION: "survey:io-end",
  SUPPORT_MODEL_IO_END_SUCCESS_ACTION: "support:io-end-success",

  FOCUS_TARGET_LAYER_ACTION: "focus",
  UNFOCUS_MODEL_ACTION: "unfocus",

  DEFAULT_DATE_FORMAT: "YYYY-MM-DD",
  DEFAULT_DATE_DISPLAY_FORMAT: "MMMM Do YYYY",

  MAP_TRANSITION_FIT_LINESTRING_COORDS: "map:fit-linestring-coords",
  MAP_TRANSITION_FIT_POLYGON_COORDS: "map:fit-polygonal-coords",
  MAP_TRANSITION_EASE_TO_POINT: "map:ease-to-point",
};
