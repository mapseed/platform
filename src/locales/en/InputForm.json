{
  "validationHeader": "Your post is looking good, but we need some more information before we can proceed."
}
