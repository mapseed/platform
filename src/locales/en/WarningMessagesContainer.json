{
  "missingGeometry": "Please use the drawing toolbar to add a point, line, or polygon to the map.",
  "missingRequired": "Please fill out the field(s) outlined below.",
  "mapNotDragged": "It looks like you haven't set a location for your post. Please drag and zoom the map to set a location."
}
