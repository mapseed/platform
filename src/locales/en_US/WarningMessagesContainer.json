{
  "missingGeometry": "Please use the drawing toolbar to add a point, line, or polygon to the map.",
  "missingRequired": "Please fill out the field(s) outlined below.",
  "duplicateUrl": "Please choose a different URL for your post. The URL you chose is already in use.",
  "mapNotDragged": "It looks like you haven't set a location for your post. Please drag and zoom the map to set a location."
}
