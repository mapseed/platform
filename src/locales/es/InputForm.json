{
  "InputForm": {
    "validationHeader": "Su publicación se ve bien, pero necesitamos más información antes de poder continuar.",
    "privateSubmissionModalHeader": "Gracias!",
    "privateSubmissionModalBody": "Su información fue enviada exitosamente y permanecerá privada."
  }
}
