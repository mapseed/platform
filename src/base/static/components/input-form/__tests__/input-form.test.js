import React from "react";
import { shallow } from "enzyme";
import InputForm from "../index.js";
import FormField from "../../form-fields/form-field";
import constants from "../../../constants";
import WarningMessagesContainer from "../../ui-elements/warning-messages-container";
import { OrderedMap, Map } from "immutable";

jest.mock("../../../utils/scroll-helpers");

describe("InputForm", () => {
  // TODO: consider generalizing this stub into a mock:
  const eventStub = { preventDefault: () => {} };
  const defaultProps = {
    containers: {},
    hideCenterPoint: () => {},
    hideSpotlightMask: () => {},
    selectedCategory: "someCategory",
    map: {
      on: () => {},
      off: () => {},
      getCenter: () => {},
    },
    places: {},
    router: {},
    showNewPin: () => {},
  };
  test("renders input form", () => {
    const wrapper = shallow(<InputForm {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders form validation errors", () => {
    const wrapper = shallow(<InputForm {...defaultProps} />);
    wrapper.setState({
      formValidationErrors: new Set(["asdf", "asdf2"]),
    });

    expect(wrapper.find(WarningMessagesContainer)).toHaveLength(1);
  });

  test("renders form fields", () => {
    const wrapper = shallow(<InputForm {...defaultProps} />);
    expect(wrapper.find(FormField)).toHaveLength(3);
  });

  test("onSubmit creates form validation errors", () => {
    const fieldName1 = "test1";
    const fieldName2 = "test2";
    const fieldName3 = "test3";
    const props = {
      ...defaultProps,
    };

    const wrapper = shallow(<InputForm {...props} />);
    const errorMessage1 = "error1";
    const errorMessage2 = "error2";
    const errorMessage3 = "error3";
    wrapper.setState({
      fields: OrderedMap({
        [fieldName1]: Map({
          [constants.FIELD_VISIBILITY_KEY]: true,
          [constants.FIELD_VALIDITY_KEY]: true,
          [constants.FIELD_VALIDITY_MESSAGE_KEY]: errorMessage1,
        }),
        [fieldName2]: Map({
          [constants.FIELD_VISIBILITY_KEY]: true,
          [constants.FIELD_VALIDITY_KEY]: true,
          [constants.FIELD_VALIDITY_MESSAGE_KEY]: errorMessage2,
        }),
        [fieldName3]: Map({
          [constants.FIELD_VISIBILITY_KEY]: true,
          [constants.FIELD_VALIDITY_KEY]: false,
          [constants.FIELD_VALIDITY_MESSAGE_KEY]: errorMessage3,
        }),
      }),
    });

    wrapper.instance().onSubmit(eventStub);
    expect(wrapper.state("formValidationErrors")).toEqual(
      new Set([errorMessage3]),
    );
  });

  test("multi stage forms render fields correctly", () => {
    const categoryName = "someMultiStageCategory";
    const fieldName1 = "test1";
    const fieldName2 = "test2";
    const fieldName3 = "test3";
    const fieldName4 = "test4";
    const props = {
      ...defaultProps,
      selectedCategory: categoryName,
    };

    const wrapper = shallow(<InputForm {...props} />);
    wrapper.setState({
      fields: OrderedMap({
        [fieldName1]: Map({
          [constants.FIELD_VISIBILITY_KEY]: true,
          [constants.FIELD_VALUE_KEY]: "",
          [constants.FIELD_VALIDITY_KEY]: true,
        }),
        [fieldName2]: Map({
          [constants.FIELD_VISIBILITY_KEY]: true,
          [constants.FIELD_VALUE_KEY]: "",
          [constants.FIELD_VALIDITY_KEY]: true,
        }),
        [fieldName3]: Map({
          [constants.FIELD_VISIBILITY_KEY]: true,
          [constants.FIELD_VALUE_KEY]: "",
          [constants.FIELD_VALIDITY_KEY]: true,
        }),
        [fieldName4]: Map({
          [constants.FIELD_VISIBILITY_KEY]: true,
          [constants.FIELD_VALUE_KEY]: "",
          [constants.FIELD_VALIDITY_KEY]: true,
        }),
      }),
    });

    expect(wrapper.find(FormField)).toHaveLength(3);
    wrapper.setState({
      currentStage: 2,
    });
    expect(wrapper.find(FormField)).toHaveLength(1);
  });
});
