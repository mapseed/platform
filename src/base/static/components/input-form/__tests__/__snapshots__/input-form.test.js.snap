// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm renders input form 1`] = `
<div
  className="input-form"
>
  <WarningMessagesContainer
    errors={Array []}
    headerMsg="validationHeader"
    t={[Function]}
  />
  <form
    className="input-form__form"
    id="mapseed-input-form"
    onSubmit={[Function]}
  >
    <FormField
      attachmentModels={Immutable.List []}
      categoryConfig={
        Object {
          "fields": Array [
            Object {
              "name": "test1",
            },
            Object {
              "name": "test2",
            },
          ],
        }
      }
      disabled={false}
      fieldConfig={
        Object {
          "name": "test1",
        }
      }
      fieldState={
        Immutable.Map {
          "value": "",
          "renderKey": 0,
        }
      }
      key="0"
      map={
        Object {
          "getCenter": [Function],
          "off": [Function],
          "on": [Function],
        }
      }
      mapConfig={Object {}}
      onAddAttachment={[Function]}
      onFieldChange={[Function]}
      onGeometryStyleChange={[Function]}
      places={Object {}}
      router={Object {}}
      showValidityStatus={false}
      t={[Function]}
      updatingField={null}
    />
    <FormField
      attachmentModels={Immutable.List []}
      categoryConfig={
        Object {
          "fields": Array [
            Object {
              "name": "test1",
            },
            Object {
              "name": "test2",
            },
          ],
        }
      }
      disabled={false}
      fieldConfig={
        Object {
          "name": "test2",
        }
      }
      fieldState={
        Immutable.Map {
          "value": "",
          "renderKey": 0,
        }
      }
      key="0"
      map={
        Object {
          "getCenter": [Function],
          "off": [Function],
          "on": [Function],
        }
      }
      mapConfig={Object {}}
      onAddAttachment={[Function]}
      onFieldChange={[Function]}
      onGeometryStyleChange={[Function]}
      places={Object {}}
      router={Object {}}
      showValidityStatus={false}
      t={[Function]}
      updatingField={null}
    />
  </form>
</div>
`;
